aws/resource_aws_gamelift_build_test.go

```diff
@@ -0,0 +1,302 @@
+	"github.com/hashicorp/terraform/helper/acctest"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+func init() {
+	resource.AddTestSweepers("aws_gamelift_build", &resource.Sweeper{
+		Name: "aws_gamelift_build",
+		F:    testSweepGameliftBuilds,
+	})
+}
+
+func testSweepGameliftBuilds(region string) error {
+	client, err := sharedClientForRegion(region)
+	if err != nil {
+		return fmt.Errorf("error getting client: %s", err)
+	}
+	conn := client.(*AWSClient).gameliftconn
+
+	resp, err := conn.ListBuilds(&gamelift.ListBuildsInput{})
+	if err != nil {
+		return fmt.Errorf("Error listing Gamelift Builds: %s", err)
+	}
+
+	if len(resp.Builds) == 0 {
+		log.Print("[DEBUG] No Gamelift Builds to sweep")
+		return nil
+	}
+
+	log.Printf("[INFO] Found %d Gamelift Builds", len(resp.Builds))
+
+	for _, build := range resp.Builds {
+		if !strings.HasPrefix(*build.Name, "tf_acc_build_") {
+			continue
+		}
+
+		log.Printf("[INFO] Deleting Gamelift Build %q", *build.BuildId)
+		_, err := conn.DeleteBuild(&gamelift.DeleteBuildInput{
+			BuildId: build.BuildId,
+		})
+		if err != nil {
+			return fmt.Errorf("Error deleting Gamelift Build (%s): %s",
+				*build.BuildId, err)
+		}
+	}
+
+	return nil
+}
+
+func TestAccAWSGameliftBuild_basic(t *testing.T) {
+	var conf gamelift.Build
+
+	rString := acctest.RandString(8)
+
+	buildName := fmt.Sprintf("tf_acc_build_%s", rString)
+	uBuildName := fmt.Sprintf("tf_acc_build_updated_%s", rString)
+	bucketName := fmt.Sprintf("tf-acc-bucket-gamelift-build-%s", rString)
+	roleName := fmt.Sprintf("tf_acc_role_%s", rString)
+	policyName := fmt.Sprintf("tf_acc_policy_%s", rString)
+
+	roleArnRe := regexp.MustCompile(":role/" + roleName + "$")
+	zipPath := "test-fixtures/gamelift-gomoku-build-sample.zip"
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAWSGameliftBuildDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSGameliftBuildBasicConfig(buildName, bucketName, zipPath, roleName, policyName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSGameliftBuildExists("aws_gamelift_build.test", &conf),
+					resource.TestCheckResourceAttr("aws_gamelift_build.test", "name", buildName),
+					resource.TestCheckResourceAttr("aws_gamelift_build.test", "operating_system", "WINDOWS_2012"),
+					resource.TestCheckResourceAttr("aws_gamelift_build.test", "storage_location.#", "1"),
+					resource.TestCheckResourceAttr("aws_gamelift_build.test", "storage_location.0.bucket", bucketName),
+					resource.TestCheckResourceAttr("aws_gamelift_build.test", "storage_location.0.key", "tf-acc-test-gl-build.zip"),
+					resource.TestMatchResourceAttr("aws_gamelift_build.test", "storage_location.0.role_arn", roleArnRe),
+				),
+			},
+			{
+				Config: testAccAWSGameliftBuildBasicUpdateConfig(uBuildName, bucketName, zipPath, roleName, policyName),
```

Comment:
Nitpick: I don't think we need the "UpdateConfig" since you've already parameterized the original configuration. 😄